### DISCUSSION TITLE
Deduping database notifications

### DISCUSSION THREAD

**Discussion by redbastie** [laravel/framework#33677] [General]
Right now there is no mechanism for deduping of database notifications in an optimized way.

Sure, you could encode an array as JSON and compare it to the `data` column, but this rules out indexing and is sure to be slow with any decent sized notifications table.

I propose we add a column we can use for deduping. Perhaps a type of hash like `base64_encode` which represents the `data`, that is actually indexable so that it can be used for deduping purposes.

A real world example of this would be a `like` feature within a web app or API. As it stands, with no deduping, a user can simply toggle their `like` on a post and it will create endless notifications.

The current database notification system is more ideal for things like messaging systems, but not the example I just mentioned.

**Comment by netpok** [comment]
I'm not sure that this is the job of the database notification driver as it applies to all the channel. In your example the user would get endless emails to.

Also I would use some kind of hash instead of base64 so the length of that is constant.
